EIPS/eip-7377.md

```diff
@@ -0,0 +1,127 @@
+---
+eip: 7377
+title: Migration Transaction
+description: Allow EOAs to send a one-time transaction which deploys code at their account.
+author: lightclient (@lightclient), Sam Wilson (@samwilsn), Ansgar Dietrichs (@adietrichs)
+discussions-to: https://ethereum-magicians.org/t/eip-xxxx-migration-transaction/15144
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-07-21
+requires: 170, 1559, 2718
+---
+
+## Abstract
+
+Introduce a new [EIP-2718](./eip-2718.md) transaction type with the format `0x04 || rlp([chainId, nonce, maxFeePerGas, maxPriorityFeePerGas, gasLimit, codePtr, storage, data, value, accessList, yParity, r, s])` which sets the sending account's `code` field in the state trie to the `code` value at `codePtr` and applies the storage tuples to the sender's storage trie.
+
+## Motivation
+
+For as long as Ethereum has existed, so have complaints about the user experience of the platform.
+
+Ethereum was designed to be an abstract platform, limited only by the mind of the developers who deploy to it. Smart contract wallets were expected to placate unhappy users. Unfortunately, many years into this journey, the tension is still here.
+
+The ecosystem looks different than it did when when the first smart contract wallets were written. Several application-level standards have mitigated some of the issues. There are large standardization efforts around smart contract wallets, and a big push to make them the default on layer two chains. Everything seems to be moving in the right direction.
+
+One significant problems that hasn't been resolved though is what to do about the EOA users of *today*. Manually migrating assets from an EOA is not a tenable path to converting Ethereum's user base to smart contract wallets. We must provide a mechanism, embedded in the protocol, to migrate. This EIP proposes such mechanism.
+
+## Specification
+
+At the fork block `X`, introduce the MigrationTransaction transaction type.
```

Comment:
Might be worth including a parameters table for the block number and transaction type, in case somehow a new Core EIP comes in the interim and takes up `0x04`.

Reply:
meh I can update it